src/app/calendar/month.cpp

```diff
@@ -0,0 +1,71 @@
+#include "month.hpp"
+#include "day.hpp"
+
+Month::Month(const std::chrono::month& month) : month_{month}
+{
+}
+
+std::chrono::month Month::getMonth()
+{
+    return month_;
+}
+
+size_t Month::getNumberOfDays()
+{
+    if ((static_cast<unsigned>(month_) % 2) == 0)
+    {
+        if (static_cast<unsigned>(month_) == 2)
+        {
+            if (year_->getYear().is_leap())
+            {
+                return 29;
+            }
+            else
+            {
+                return 28;
+            }
+        }
+        else
+        {
+            return 30;
+        }
+    }
+    else
+    {
+        return 31;
+    }
+}
+
+void Month::addYear(const std::shared_ptr<Year>& year)
+{
+    year_ = year;
+}
+
+void Month::generateDays(const DayName& day_name)
+{
+    size_t days_in_month = getNumberOfDays();
+    auto month_ptr = shared_from_this();
+
+    DayName next_day_name;
+    if (static_cast<unsigned>(getMonth()) == 1)
+    {
+        next_day_name = day_name;
+    }
+    else
+    {
+        next_day_name = day_name;
+        ++next_day_name;
+    }
+
+    for (size_t i = 1; i <= days_in_month; ++i)
+    {
+        auto day = std::make_shared<Day>(std::chrono::day{static_cast<unsigned>(i)}, next_day_name);
+        day->addMonth(month_ptr);
+        days_.push_back(day);
+    }
+}
```

Comment:
I think generation should be in Calendar. 
From a given point: DD::MM::YYYY we should start creating objects.
Let's call it ymd (from std::chrono::year_month_day) 
while(???) 
{
if(ymd !=  std::chrono::year_month_day_last) 
{
createDayOfMonth(ymd.month()); - here day of ymd should be incremented 
and then check again. 
There should be also check for last day of month, if it's the last one, then month should be incremented and day set to 1st of the month.
}}

Reply:
This way we could handle it by vector of pairs 
`vector<pair<YMD, our_implementation_of_day>> `
And then we will have only Day class, which will have our implementation of timesheet for company.

Reply:
One more thing came to my mind. 
This way we will have multiple objects ready and waiting, but most of them won't be ever filled. 

Wouldn't it be better to have some "map" that will contain only data such as date until action on such date is not taken?
Thinking of performance it might be needed to be done like this, otherwise we will have a thousands of object which are initialized but empty. 

Then once we will have some visit at some day, the day will be created and linked. Though tho. It might be next step for calendar.

Reply:
Another thing is how should previous days look like. 
Those days are no longer useful, they only contain data about what happen at what time. 
Maybe it should be simplified to other data type, that will contain only timestamps that were used. 
For example. 
Yesterday we had 2 visits, 9:30-10:30 and 14:00-17:00
So only timestamps from 9:30 to 10:30 and 14:00 to 17:00 are in this object, the rest is cut. Hmm